lib/msf/base/sessions/command_shell.rb

```diff
@@ -312,18 +312,18 @@ def cmd_shell(*args)
   end
 
   #
-  # Check if there is a binary in PATH env
+  # Returns path of a binary in PATH env.
   #
   def binary_exists(binary)
     print_status("Trying to find binary(#{binary}) on target machine")
-    binary_path = shell_command_token("which #{binary}").to_s.strip
-    if binary_path.eql?("#{binary} not found") || binary_path == ""
-      print_error(binary_path)
+    binary_path = shell_command_token("command -v '#{binary}' && echo true")
+    if binary_path.eql?("")
+      print_error("#{binary} not found")
       return nil
-    else
-      print_status("Found #{binary} at #{binary_path}")
-      return binary_path
     end
+    binary_path=binary_path.split[0].chomp
```

Comment:
What is the purpose of this line?

It is splitting by whitespace and will drop anything after the first space which doesn't seem right.

```suggestion
    binary_path = binary_path.split[0].chomp
```

Reply:
Yeah, missed `("\n")`, will change that. This lines removes the `true` that will get printed as a result of `&& echo true` on the next line, and gets the actual path.